kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/UploadSourceTask.kt

```diff
@@ -0,0 +1,110 @@
+package org.radarbase.connect.upload
+
+import org.apache.kafka.connect.errors.ConnectException
+import org.apache.kafka.connect.source.SourceRecord
+import org.apache.kafka.connect.source.SourceTask
+import org.radarbase.connect.upload.converter.Converter
+import org.radarbase.connect.upload.api.PollDTO
+import org.radarbase.connect.upload.api.RecordMetadataDTO
+import org.radarbase.connect.upload.api.UploadBackendClient
+import org.radarbase.connect.upload.converter.Converter.Companion.END_OF_RECORD_KEY
+import org.radarbase.connect.upload.converter.Converter.Companion.RECORD_ID_KEY
+import org.radarbase.connect.upload.converter.Converter.Companion.REVISION_KEY
+import org.radarbase.connect.upload.converter.altoida.AltoidaCsvExportConverter
+import org.radarbase.connect.upload.util.VersionUtil
+import org.slf4j.LoggerFactory
+import java.lang.Exception
+
+class UploadSourceTask: SourceTask() {
+    private lateinit var uploadClient: UploadBackendClient
+    private lateinit var converters: List<Converter>
+
+    override fun start(props: Map<String, String>?) {
+
+        val connectConfig = UploadSourceConnectorConfig(props!!)
+        uploadClient = UploadBackendClient(
+                connectConfig.getAuthorizer(),
+                connectConfig.getHttpClient(),
+                connectConfig.getBaseUrl())
+
+        // init default converters
+        converters = mutableListOf(AltoidaCsvExportConverter())
+
+        // init additional converters if configured
+        val additionalConverters = connectConfig.getConverterClasses().map {
+            try {
+                val converterClass = Class.forName(it)
+                converterClass.getDeclaredConstructor().newInstance() as Converter
+            } catch (exe: Exception) {
+                when (exe) {
+                    is ClassNotFoundException -> throw ConnectException("Converter class ${it} not found in class path", exe)
+                    is IllegalAccessException -> throw ConnectException("Converter class ${it} could not be instantiated", exe)
+                    is InstantiationException -> throw ConnectException("Converter class ${it} could not be instantiated", exe)
+                    else -> throw ConnectException("Cannot successfully initialize converter ${it}", exe)
+                }
+            }
+        }.toList()
+
+        converters.plus(additionalConverters)
```

Comment:
FYI: this does not modify the converters list, so this operation does not do anything. Either assign converters in a single statement, or make it a `MutableList` and use `addAll`.